srcjs/input_binding_date.js

```diff
@@ -16,10 +16,17 @@ $.extend(dateInputBinding, {
   setValue: function(el, value) {
     var date = this._newDate(value);
     // If date is invalid, do nothing
+    // Note: R's NA is null here, and isNaN(null) is false
```

Comment:
Sorry for being unclear -- I meant checking `value === null` right at the top of this function.